### ISSUE TITLE
CORS should allow `*.operationcode.org` so the /resources page can get resources

### ISSUE THREAD

**Issue by aaron-suarez** [OperationCode/resources_api#250]
Per the conversation in this issue https://github.com/OperationCode/front-end/issues/661, the lack of Access Control headers is preventing the front end from getting the resources from the API. We need to add CORS headers, but we should only open it up to the Operation Code frontend.

**Comment by kylemh** [comment]
and local host (port 3000), and also probably any of our potential Now aliases!

`*.operation-code.now.sh`
`operation-code.now.sh`

**Comment by aaron-suarez** [comment]
@kylemh I'm not familiar with Now, so I have to ask the question. Are all of those subdomains fully allocated to us from Now? For example, can someone who's not a part of Operation Code go get `operation-code-not-really-oc.now.sh`?

**Comment by kylemh** [comment]
That is indeed possible! It's okay though, because you'll require on authentication token for POSTs and PUTs and DELETEs so that only GETs are valid for any of these potential subdomain thieves!

This isn't a limitation for Now, it's just that we're stuck using this system because we agreed to keep using Terraform / k8s for the root host.

**Comment by chynh** [comment]
Hey all, any progress on this issue? 

I am asking because I did some work on ``resources`` page which involves making GET requests to resources API. I can confirm the change works with resources API on localhost but I thought it would be safe to check in on this issue before going forward.

**Comment by kylemh** [comment]
@chynh the page already exists! You should check out master.

**Comment by chynh** [comment]
@kylemh Yes. I am actually working off of master.

Basically, I am trying to make it so that there are fewer resources displayed at a time and allow users to access resources other than the first twenty which are what they get right now.

In doing so, I have modified the URL from ``https://resources.operationcode.org/api/v1/search?page=1`` to ``https://resources.operationcode.org/api/v1/search?page=PAGE_NUM&page_size=NUM_OF_RESOURCES_TO_DISPLAY`` but it only works when I set the baseURL to localhost.

**Comment by kylemh** [comment]
Hmmm so where is it not workin?

https://operation-code.now.sh/resources seems to yield data

**Comment by chynh** [comment]
So I have made the Pagination component clickable so that you would make a new request to the API for a specific page and that's where it won't work.

Here is the link to the video for your reference:
https://imgur.com/4fP7rFj

**Comment by aaron-suarez** [comment]
I think the main issue was whether we can make POST and PUT requests (we don't currently have a DELETE endpoint, but if we did, that would be in scope for this too).

I believe there's still some front end work to go before we'll know if POST and PUT are blocked by CORS, but I imagine they're not because I think that GET would fail as well. I'm going to pair with @paul tomorrow evening because he wanted to try to tackle this issue. You're welcome to join us on the call if you like, we'll be setting him up as a contributor. I'll start the call in the #oc-python-projects channel on slack so anyone interested can join.

Do you think 20 resources is too many?

**Comment by aaron-suarez** [comment]
Dang, I didn't mean to tag @ paul, the person I meant to tag didn't populate right 😳

**Comment by chynh** [comment]
Cool. I probably will be busy with packing my stuff up since I am moving soon but I will definitely keep on eye on slack and see if I can join in!

And the number of resources per page is definitely something that is up for a discussion. I just picked an arbitrary number to show per page. 

Although, the issue with CORS is still there even without limiting the number of resources as I just tried to see if it would work.
